### ISSUE TITLE
Program crash when parsing file

### ISSUE THREAD

**Issue by chtenb** [koka-lang/koka#379]
I'm currently doing the first challenge of advent of code, and I'm hitting a bug in Koka causing my program to segfault during parsing the input file. The smaller example input file works fine, but when parsing the larger input file the program hangs halfway for a few seconds, then aborts.

I'm on Windows using the lastest version of the Koka compiler. How can I help with providing debug logs etc?

**Comment by TimWhiting** [comment]
You'll quickly run out of stack space with the builtin parse function. It induces exponential backtracking, and parsing a single character at a time will make that even worse. On linux you can request `infinite` stack space using `ulimit -s unlimited`. I'm not sure if there is an equivalent windows command.

**Comment by daanx** [comment]
Yes, the current builtin parser is not great -- we should improve it. For now it is mostly there to avoid a regex engine dependency when parsing integers and floats in the standard libraries.
